### ISSUE TITLE
Remove `away` group

### ISSUE THREAD

**Issue by lukasjuhrich** [agdsn/pycroft#90]
In times of older finance regulations, there was a reduced membership consisting only of email.
It is correct that users will be added to these groups by the importer if a reduced fee has been booked.

However, this group shouldn't be used in the _current_ library anymore.

Thus, at least `lib.user.move_out_temporarily` should be rewritten in order to not add an `away` membership, but to add a `closedopen()` membership in `member` _after_ the given absence period ended. (Refs #46)

To be done:
- remove `away` group from config and library

**Comment by lukasjuhrich** [comment]
One side note: admins who have been moved out have to be handled correctly (by the importer, as well) –  they need to get the `member` group so they will get access to the services and are forced to pay the full fee.

**Comment by sebschrader** [comment]
Thinking of it. Maybe we should keep the group.

The case that member's will leave us temporarily remains and we might want to document that through a membership.

**Comment by lukasjuhrich** [comment]
@sebschrader The information of being away temporarily is already captured implicitly by starting to be a member in a later point in time, i.e. by the boundedness of the “empty spot” between the both membership intervals. Ever since we don't require a fee for the services anymore, its purpose has been renderd obsolete.

**Comment by sebschrader** [comment]
Point taken. I agree.

**Comment by lukasjuhrich** [comment]
# Unforeseen Problems

When working with this issue, I noticed the following behavioural mistakes:

- `move_out_temporarily` Removes the user's IPs *every time*, even if the date is in the future
- `move_out` does nothing of that kind

Note that there is not a possibility of asynchronously executing tasks (like in this case, deleting hosts/IPs). Therefore, the possibility of giving a custom date for `move_out` has to be disabled in the UI and should raise a `NotImplementedError` if `lib.user.move_out` is called accordingly (the function signature should not be changed just for a temporary feature disabling).

I talked to @lightning1 about how we should handle the removal of hosts and IPs concerning the possible moving back in of a user. If we remove the hosts immediately, the MAC has to be reprovided if the user moves back in. We came to the conclusion that it is a bearable inconvenience, because else, we would soon have a big pile of obsolete hosts with no clear cleanup procedure.

To summarize:
- [ ] Disable moving-out dates in the future
- [ ] Disable moving back in
- [ ] Always delete a user's hots and IPs
- [ ] Add a note hinting that `moving out` does not just remove the room, but also the membership, hosts and IPs.
- [ ] Add a note that people might want to be added to certain groups in order to not lose access to services. If there's time, implement a check box in the UI that automatically does that.
- [ ] For the above: Add a group granting all “services” properties (currently only `mail`) named “Access to Services” for former active and passive members, and create it in the importer, and reference this  this group in the config.

**Comment by sebschrader** [comment]
Keeping old hosts doesn't seem like a huge problem to me. Keeping old IPs on the other hand is probably a problem with the current schema, as it will prevent them from being reallocated for other users.

This is a problem, because we shouldn't delete the IP information immediately and keep it at least for a little while. Otherwise we can't give information about the identity of the user of an IP address anymore.

**Comment by mmarx** [comment]
> Otherwise we can't give information about the identity of the user of an IP address anymore.

Why is that a problem?

**Comment by sebschrader** [comment]
We should be able to this for at least a few days after some ends its membership.

**Comment by sebschrader** [comment]
I don't think it's wise to get into a legal discussion here on GitHub. We should take this some place else.
